syntax = "proto3";

service FrbSifter {

  // First call - report my configuration.
  // Returns true if good, false otherwise.
  rpc CheckConfiguration (ConfigMessage) returns (ConfigReply) {}

  // ?? For injections only
  //rpc NewBeamSet (BeamSetUpdateMessage) returns (BeamSetUpdateReply) {}

  // Report events from a given time chunk.
  rpc FrbEvents (FrbEventsMessage) returns (FrbEventsReply) {}

}

message ConfigMessage {
  string yaml = 1;
}

message ConfigReply {
  bool ok = 1;
}

/*
* message BeamSetUpdateMessage {
*   int32 beam_set_id = 1;
*   repeated int32 beam = 2;
* }
* message BeamSetUpdateReply {
*   bool = 1;
* }
*/

message FrbEvent {
  int32 beam_id = 1;
  int64 fpga_timestamp = 2;
  float dm = 3;
  float dm_error = 4;
  float snr = 5;
  float rfi_prob = 6;
}

message FrbEventsMessage {
  bool has_injections = 1;
  int32 beam_set_id = 2;
  int64 chunk_fpga_count = 3;
  repeated FrbEvent events = 4;
}

message FrbEventsReply {
  bool ok = 1;
}

